### PR TITLE
Update snow DA cycling for the fractional grid setting.

### PR DESCRIPTION
**Pull request by jiaruidong2017**
This PR will allow users to run JEDI-based LETKFOI update for snow depth analysis w/o the fractional grids settings of GFSv17. 

In previous GFS versions, the variable for total snow depth (`snwdph`) was only defined over land grid cells. In the new GFSv17 restarts, with fractional grids, the `snwdph` variable in the restarts is the average snow depth over the grid cell [ = area-weighted average of the snow depth over sea ice (`snodi`) and over land (`snodl`)]. `snodl` is the snow depth over land portion of a cell, `snodi` is the snow depth over ice portion of a cell, and `snwdph` is the snow depth over the entire grid cell. 

This PR depend on:
https://github.com/NOAA-EMC/GDASApp/pull/504

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

Ran a standalone landanl.sh job on Hera.
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: sorc/link_workflow.sh
Comment (aerorahul) [inline sorc/link_workflow.sh]: Why is a yaml file being linked into `ush`?

File: parm/parm_gdas/land_jedi_fix_gfsv17.yaml
Comment (aerorahul) [inline parm/parm_gdas/land_jedi_fix_gfsv17.yaml]: Why is this in `ush`?  This should be a file in `$(HOMEgfs)/parm/parm_gdas/...`

File: ush/python/pygfs/task/land_analysis.py
Comment (aerorahul) [inline ush/python/pygfs/task/land_analysis.py]: I think we can do this in the `land_jedi_fix.yaml` itself instead of in the python script.

File: parm/parm_gdas/land_jedi_fix_gfsv17.yaml
Comment (aerorahul) [reply parm/parm_gdas/land_jedi_fix_gfsv17.yaml]: I think we can handle this case in the `land_jedi_fix.yaml`  itself instead of adding this yaml.

File: .gitignore
Comment (aerorahul) [inline .gitignore]: why is a yaml in a `ush`?

**Review by aerorahul — CHANGES_REQUESTED** [review]
Please see inline comments.

**Comment by jiaruidong2017** [comment]
Thanks @aerorahul for your quick review. I did agree with all you suggestions. 

However, we just had a meeting and plan to handle the fractional grid issue in JEDI fv3-jedi repo directly. Therefore, I will close this PR. Thanks again.

**Comment by jiaruidong2017** [comment]
@aerorahul @CoryMartin-NOAA I have a concern. The current global-workflow supports both fractional grid cells and non fractional grid cells. If the above proposed changes are made in the JEDI fv3-jedi repo, the snow DA cycling will not work any more for the non fractional grid settings.

**Comment by aerorahul** [comment]
> @aerorahul @CoryMartin-NOAA I have a concern. The current global-workflow supports both fractional grid cells and non fractional grid cells. If the above proposed changes are made in the JEDI fv3-jedi repo, the snow DA cycling will not work any more for the non fractional grid settings.

@jiaruidong2017 
global-workflow does not support non-fractional grid IIRC.  v17 will use fractional grid to my knowledge.

**Comment by CoryMartin-NOAA** [comment]
So is this PR able to be closed then?

**Comment by jiaruidong2017** [comment]
As I mentioned in GDASApp's PR, the proposed changes can support both fractional and non fractional grids in the global-workflow.

**Comment by CoryMartin-NOAA** [comment]
@jiaruidong2017 is that true for the model or just the DA?

**Comment by aerorahul** [comment]
The workflow no longer supports non-fractional grid.

**Comment by jiaruidong2017** [comment]
@CoryMartin-NOAA I mean for snow DA only

**Comment by CoryMartin-NOAA** [comment]
@jiaruidong2017 if the model doesn't support it, then there is no real need for us to support both for the snow DA

**Comment by jiaruidong2017** [comment]
@aerorahul Do you mean we need to remove all `FRAC_GRID`?

**Comment by JessicaMeixner-NOAA** [comment]
The ufs-weather-model itself does support both fractional and non-fractional grid to my knowledge.  @junwang-noaa can correct if needed.

**Comment by jiaruidong2017** [comment]
@CoryMartin-NOAA I have no idea whether the model support the non fractional grid or not, and I will check this with @barlage

**Comment by jiaruidong2017** [comment]
Thanks @JessicaMeixner-NOAA for your info.

**Review by CoryMartin-NOAA — COMMENTED** [review]
FRAC_GRID is still a variable in the workflow from a search, so from what I can tell, this one line change looks good to me. But is this PR in draft waiting for a GDASApp hash?

**Comment by jiaruidong2017** [comment]
@CoryMartin-NOAA Yes, this PR depends on the [GDASApp PR #504](https://github.com/NOAA-EMC/GDASApp/pull/504)

**Comment by junwang-noaa** [comment]
Yes, the model does support both fractional grid and non-fractional grid.

**Comment by aerorahul** [comment]
The model does support non-fractional grids.
GFSv17 and GEFSv13 are only implementing fractional grids.
The global-workflow has a variable `FRAC_GRID`, however, the workflow does not support non-fractional grid configurations.  We may remove `FRAC_GRID` references from the global-workflow.
Use at your own risk.

**Comment by jiaruidong2017** [comment]
The `gfs-land.yaml` in the fix `fv3jedi` directory need to be updated for the snow DA cycling in the global-workflow. 
@WalterKolczynski-NOAA Would you please copy the file below:  
`/scratch2/NCEPDEV/stmp1/Jiarui.Dong/gdasapp/gdasapp.9/fv3-jedi/test/Data/fieldmetadata/gfs-land.yaml`
to replace: 
`${FIX_DIR}/gdas/fv3jedi/20220805/fieldmetadata/gfs-land.yaml`

`FIX_DIR` can be found in `sorc/link_workflow.sh` for different platforms as below: 
```
  "wcoss2")   FIX_DIR="/lfs/h2/emc/global/noscrub/emc.global/FIX/fix" ;;
  "hera")     FIX_DIR="/scratch1/NCEPDEV/global/glopara/fix" ;;
  "orion")    FIX_DIR="/work/noaa/global/glopara/fix" ;;
  "jet")      FIX_DIR="/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix" ;;
  "s4")       FIX_DIR="/data/prod/glopara/fix" ;;
```

Since there are many changes in `fieldmetadata` and `fv3files`, you may want to update the `fv3jedi` link to a new date `yyyymmdd` by copy the following directories: 
```
/scratch2/NCEPDEV/stmp1/Jiarui.Dong/gdasapp/gdasapp.9/fv3-jedi/test/Data/fieldmetadata
/scratch2/NCEPDEV/stmp1/Jiarui.Dong/gdasapp/gdasapp.9/fv3-jedi/test/Data/fv3files
```
to: 
`${FIX_DIR}/gdas/fv3jedi/yyyymmdd/`

Please let me and @CoryMartin-NOAA know if you have any questions. Thanks.

**Comment by aerorahul** [comment]
> The `gfs-land.yaml` in the fix `fv3jedi` directory need to be updated for the snow DA cycling in the global-workflow. @WalterKolczynski-NOAA Would you please copy the file below: `/scratch2/NCEPDEV/stmp1/Jiarui.Dong/gdasapp/gdasapp.9/fv3-jedi/test/Data/fieldmetadata/gfs-land.yaml` to replace: `${FIX_DIR}/gdas/fv3jedi/20220805/fieldmetadata/gfs-land.yaml`
> 
> `FIX_DIR` can be found in `sorc/link_workflow.sh` for different platforms as below:
> 
> ```
>   "wcoss2")   FIX_DIR="/lfs/h2/emc/global/noscrub/emc.global/FIX/fix" ;;
>   "hera")     FIX_DIR="/scratch1/NCEPDEV/global/glopara/fix" ;;
>   "orion")    FIX_DIR="/work/noaa/global/glopara/fix" ;;
>   "jet")      FIX_DIR="/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix" ;;
>   "s4")       FIX_DIR="/data/prod/glopara/fix" ;;
> ```
> 
> Since there are many changes in `fieldmetadata` and `fv3files`, you may want to update the `fv3jedi` link to a new date `yyyymmdd` by copy the following directories:
> 
> ```
> /scratch2/NCEPDEV/stmp1/Jiarui.Dong/gdasapp/gdasapp.9/fv3-jedi/test/Data/fieldmetadata
> /scratch2/NCEPDEV/stmp1/Jiarui.Dong/gdasapp/gdasapp.9/fv3-jedi/test/Data/fv3files
> ```
> 
> to: `${FIX_DIR}/gdas/fv3jedi/yyyymmdd/`
> 
> Please let me and @CoryMartin-NOAA know if you have any questions. Thanks.

@jiaruidong2017 
There is a separate template for updating fix files.
Please open an issue using that template and provide the details for the update.

**Review by aerorahul — CHANGES_REQUESTED** [review]
@jiaruidong2017 
There are references to `FRAC_GRID` in `land_analysis.py`.  Please remove them.
Also, https://github.com/NOAA-EMC/GDASApp/blob/develop/parm/land/letkfoi/apply_incr_nml.j2 still contains a variable `FRAC_GRID`. It should be replaced to be `.true.`

**Review by aerorahul — APPROVED** [review]
Thanks for making the change @jiaruidong2017 
Should we wait for the NOAA-EMC/GDASApp#527 to be merged before merging this PR?

**Comment by jiaruidong2017** [comment]
> Thanks for making the change @jiaruidong2017 Should we wait for the [NOAA-EMC/GDASApp#527](https://github.com/NOAA-EMC/GDASApp/pull/527) to be merged before merging this PR?

I believe it depends on GDASApp#527, and we should wait for it. Thank you.

**Comment by jiaruidong2017** [comment]
This PR depends on the `merge develop to feature/stable-nightly` in GDASApp, and I will update the commit hash.

**Comment by jiaruidong2017** [comment]
@CoryMartin-NOAA The simple changes below in `link_workflow.sh` can solve my issues. 

```
<   for gdas_sub in crtm fv3jedi gsibec; do
---
>   for gdas_sub in crtm gsibec; do
176a177,180
>   if [[ -d "fv3jedi" ]]; then
>      rm -rf "fv3jedi"
>   fi
>   ${LINK_OR_COPY} "${script_dir}/gdas.cd/fv3-jedi/test/Data" "fv3jedi"
```
This change just links the `fv3jedi` to the `fv3-jedi` through GDASApp. How do you think? Thanks.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Provisionally approved pending CI tests.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Comment by jiaruidong2017** [comment]
@CoryMartin-NOAA Would you please merge `develop` into `feature/stable-nightly` in GDASApp so that I could update the commit hash  to include my recent changes? Thanks.

File: sorc/checkout.sh
Comment (aerorahul) [inline sorc/checkout.sh]: Sorry, I didn't catch this, but this hash needs to be also updated in `Externals.cfg`

**Review by aerorahul — CHANGES_REQUESTED** [review]
Apologies, but the `GDASApp` hash needs to be updated in `Externals.cfg` as well.

**Comment by CoryMartin-NOAA** [comment]
@jiaruidong2017 I cannot do that as that would break other capabilities as develop does not currently build for GDASApp due to a CRTM issue.

**Comment by CoryMartin-NOAA** [comment]
@jiaruidong2017 verify that https://github.com/NOAA-EMC/GDASApp/commit/4452b0afb088c240e34fa73af756a91cfe045491 has what you need and if so, update this PR to use this commit

**Comment by jiaruidong2017** [comment]
Thanks @CoryMartin-NOAA and I will test the https://github.com/NOAA-EMC/GDASApp/commit/4452b0afb088c240e34fa73af756a91cfe045491

File: sorc/checkout.sh
Comment (jiaruidong2017) [reply sorc/checkout.sh]: This has been updated. Thanks.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Comment by WalterKolczynski-NOAA** [comment]
Is this ready to merge?

**Review by aerorahul — APPROVED** [review]
looks good.

**Comment by aerorahul** [comment]
Assuming the hashes of `gdasapp` are correct, merging now.

**Pull request merged**
